eva-server/src/main/java/uk/ac/ebi/variation/eva/server/ws/FilesWSServer.java

```diff
@@ -22,64 +22,53 @@
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.Api;
 import uk.ac.ebi.variation.eva.lib.datastore.DBAdaptorConnector;
 import uk.ac.ebi.variation.eva.lib.storage.metadata.VariantSourceEvaproDBAdaptor;
-import uk.ac.ebi.variation.eva.server.exception.SpeciesException;
-import uk.ac.ebi.variation.eva.server.exception.VersionException;
 
 /**
  *
  * @author Cristina Yenyxe Gonzalez Garcia <cyenyxe@ebi.ac.uk>
  */
-@Path("/v1/files")
-@Produces("application/json")
+@RestController
+@RequestMapping(value = "/v1/files", produces = "application/json")
 @Api(tags = { "files" })
 public class FilesWSServer extends EvaWSServer {
 
     private final VariantSourceDBAdaptor variantSourceEvaproDbAdaptor;
 
-
-    public FilesWSServer(@Context UriInfo uriInfo, @Context HttpServletRequest hsr) throws NamingException, IOException {
-        super(uriInfo, hsr);
+    public FilesWSServer() throws NamingException, IOException {
         variantSourceEvaproDbAdaptor = new VariantSourceEvaproDBAdaptor();
+        this.startTime = System.currentTimeMillis();
     }
 
-    @GET
-    @Path("/all")
+    @RequestMapping(value = "/all", method = RequestMethod.GET)
 //    @ApiOperation(httpMethod = "GET", value = "Gets the files of a species")
-    public Response getFiles(@QueryParam("species") String species) 
+    public QueryResponse getFiles(@RequestParam("species") String species) 
             throws UnknownHostException, IllegalOpenCGACredentialsException, IOException {
-        try {
-            checkParams();
-        } catch (VersionException | SpeciesException ex) {
-            return createErrorResponse(ex.toString());
-        }
+        initializeQueryOptions();
         
         VariantSourceDBAdaptor variantSourceMongoDbAdaptor = DBAdaptorConnector.getVariantSourceDBAdaptor(species);
-        return createOkResponse(variantSourceMongoDbAdaptor.getAllSources(queryOptions));
+        return setQueryResponse(variantSourceMongoDbAdaptor.getAllSources(queryOptions));
     }
 
-    @GET
-    @Path("/{files}/url")
+    @RequestMapping(value = "/{files}/url", method = RequestMethod.GET)
 //    @ApiOperation(httpMethod = "GET", value = "Gets the URL of a file")
-    public Response getFileUrl(@PathParam("files") String filenames) {
-        try {
-            checkParams();
-        } catch (VersionException | SpeciesException ex) {
-            return createErrorResponse(ex.toString());
-        }
-        
-        return createOkResponse(variantSourceEvaproDbAdaptor.getSourceDownloadUrlByName(Arrays.asList(filenames.split(","))));
+    public QueryResponse getFileUrl(@PathVariable("files") String filenames) {
+        initializeQueryOptions();
```

Comment:
Previously it was kind of an "application level" object, which caused unexpected side effects with the object not being properly cleaned-up. So yes, ideally it should be initialized for each request.